### PR TITLE
Save email from stripe token to user (if they didn't have an email already)

### PR DESCRIPTION
**Pull request by phoenixeliot**

### CODE REVIEW

File: app/models/User.coffee [hunk 1]
```diff
@@ -429,7 +429,11 @@ module.exports = class User extends CocoModel
     stripe.planID = 'basic'
     stripe.token = token.id
     @set({stripe})
-    return me.patch({headers: {'X-Change-Plan': 'true'}})
+    return me.patch({headers: {'X-Change-Plan': 'true'}}).then =>
+      unless utils.isValidEmail(@get('email'))
+        @set({email: token.email})
+        me.patch()
+      return Promise.resolve()
```
Comment (sderickson) [inline app/models/User.coffee]: This alters what the Promise returns. Looks like this won't have any effect, and will be replaced with API, so no change needed, but something to be aware of.

File: server/handlers/user_handler.coffee [hunk 2]
```diff
@@ -110,6 +110,15 @@ UserHandler = class UserHandler extends Handler
...
       return callback(null, req, user) unless req.body.email?
       
+      emailRegex = /[A-z0-9._%+-]+@[A-z0-9.-]+\.[A-z]{2,63}/
+      if not emailRegex.test(req.body.email) and emailRegex.test(user.get('email'))
+        # Don't let them remove their email address if it's there already
+        # TODO: Send a response that the user can actually see! Mimic schema error?
+        if not user.get('role')
+          return callback({ res: { message: 'Individual accounts must have a valid email address', code: 422}, code: 422 })
+        if user.get('role') is 'teacher'
+          return callback({ res: { message: 'Teacher accounts must have a valid email address', code: 422}, code: 422 })
```
Comment (sderickson) [inline server/handlers/user_handler.coffee]: This should use `user.isTeacher()`. Users have other roles besides just 'teacher' which should be included.

File: spec/server/functional/user.spec.coffee [hunk 3]
```diff
@@ -218,17 +218,17 @@ describe 'PUT /db/user', ->
...
     
-  it 'allows unsetting email, even when there\'s a user with email and emailLower set to empty string', utils.wrap ->
+  it 'allows unsetting email on student accounts, even when there\'s a user with email and emailLower set to empty string', utils.wrap ->
     invalidUser = yield utils.initUser()
     yield invalidUser.update({$set: {email: '', emailLower: ''}})
-    user = yield utils.initUser()
+    user = yield utils.initUser({role: 'student'})
     yield utils.loginUser(user)
     [res, body] = yield request.putAsync { uri: getURL('/db/user/'+user.id), json: { email: '' }}
     expect(res.statusCode).toBe(200)
```
Comment (sderickson) [inline spec/server/functional/user.spec.coffee]: Should add a test here for making sure when you have an email, then remove it as an individual, you get a 422.

File: app/models/User.coffee (hunk 1 above)
Comment (phoenixeliot) [inline app/models/User.coffee]: Would the way around that to use this?
```.then (value) =>
...
return value
```

Comment (basicer) [inline app/models/User.coffee]: Yup :)

**Pull request closed without merging**